index.js

```diff
@@ -25,13 +25,31 @@ var UNICODE_CHARS = {
     '\u2029': '\\u2029'
 };
 
+// We can‘t just instanceof Date since dates are already converted to strings
+// because of native Date.prototype.JSON (which use toISOString)
+var DATE_LENGTH = new Date().toISOString().length
+function isDate(d) {
+  try {
+    return (
+      // testing length first to avoid new Date() for every string
+      d.length === DATE_LENGTH &&
```

Comment:
@caridy I don't understand why the the date length check (which for ISO date strings is always `length == 24`), is a bad heuristic to then pass the value into the date constructor and try to see if you get back out the same string?

```js
function isSerializedDate(potentialDateString) {
  if (potentialDateString.length === 24) {
    let d = new Date(potentialDateString);
    return isFinite(d) && d.toISOString() === potentialDateString;
  }

  return false;
}
```